src/core/Akka.Streams.Tests/Dsl/AsyncEnumerableSpec.cs

```diff
@@ -269,6 +269,21 @@ async IAsyncEnumerable<int> GenerateInts()
             });
         }
 
+        [Fact]
+        public async Task AsyncEnumerableSource_Disposes_OnCancel()
+        {
+            var resource = new Resource();
+            var tcs = new System.Threading.Tasks.TaskCompletionSource<NotUsed>(TaskCreationOptions
+                .RunContinuationsAsynchronously);
+            var src = Source.From(() =>
+                CancelTestGenerator(tcs, resource, default));
+            src.To(Sink.Ignore<int>()).Run(Materializer);
+            await tcs.Task;
+            Materializer.Shutdown();
+            await Task.Delay(500);
+            Assert.False(resource.IsActive);
+        }
+
```

Comment:
Adapted from #6903 report.
Calling `Materializer.Shutdown()` is, as far as I know, going to happen when the actorsystem shuts down. 

OTOH this test was added in a red->green refactoring, so I know without the changes in `AsyncEnumerable` it fails.